src/Service/QuapService.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace App\Service;
+
+use App\DTO\Mapper\QuapQuestionnaireMapper;
+use App\Entity\Group;
+use App\Entity\WidgetQuap;
+use App\Repository\QuestionnaireRepository;
+use App\Repository\WidgetQuapRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use http\Exception\BadMessageException;
+
+class QuapService
+{
+    /**
+     * @var QuestionnaireRepository $questionnaireRepository
+     */
+    private $questionnaireRepository;
+
+    /**
+     * @var WidgetQuapRepository $quapRepository
+     */
+    private $quapRepository;
+
+    /**
+     * @var EntityManagerInterface $em
+     */
+    private $em;
+
+    /**
+     * @param QuestionnaireRepository $questionnaireRepository
+     * @param WidgetQuapRepository $quapRepository
+     * @param EntityManagerInterface $em
+     */
+    public function __construct(QuestionnaireRepository $questionnaireRepository, WidgetQuapRepository $quapRepository, EntityManagerInterface $em)
+    {
+        $this->questionnaireRepository = $questionnaireRepository;
+        $this->quapRepository = $quapRepository;
+        $this->em = $em;
+    }
+
+
+    public function getQuestionnaireDataByType(string $type, string $locale, \DateTime $dateTime): array
+    {
+
+        $questionnaires = $this->questionnaireRepository->findBy([
+            "type" => $type
+        ]);
+
+        $aspects = QuapQuestionnaireMapper::createFromEntities($questionnaires, $locale, $dateTime);
```

Comment:
you should use a mapper for only one entity here. The type attribute of the questionnaire is unique